src/submodule.c

```diff
@@ -1916,31 +1921,32 @@ static int submodule_load_from_wd_lite(git_submodule *sm)
 
 /**
  * Returns a snapshot of $WORK_TREE/.gitmodules.
- *
- * We ignore any errors and just pretend the file isn't there.
  */
-static git_config *gitmodules_snapshot(git_repository *repo)
+static int gitmodules_snapshot(git_config **snap, git_repository *repo)
 {
 	const char *workdir = git_repository_workdir(repo);
-	git_config *mods = NULL, *snap = NULL;
-	git_buf path = GIT_BUF_INIT;
 
 	if (workdir != NULL) {
+		int error;
+		git_buf path = GIT_BUF_INIT;
+		git_config *mods = NULL;
 		if (git_buf_joinpath(&path, workdir, GIT_MODULES_FILE) != 0)
-			return NULL;
+			return 0;
```

Comment:
Hum. Shouldn't this bubble up the error? `git_buf_joinpath` failing indicates that there was a memory allocation error, which definitly shouldn't be ignored.

```
if (git_buf_joinpath(&path, workdir, GIT_MODULES_FILE) < 0)
    return -1;
```